### DISCUSSION TITLE
Dell Precision T3620 Resizable Bar

### DISCUSSION THREAD

**Discussion by SternXD** [xCuri0/ReBarUEFI#211] [Q&A]
Hello, I'm looking at trying to enable Resizable BAR on my computer I was able to enable Above 4G Decoding using that mod grub tool, but I'm stuck at adding it to the volume does anyone have any suggestions?

**Comment by SternXD** [comment]
Could you possibly help me? @xCuri0

**Comment by SternXD** [comment]
Alright so I was able to get further I'm stuck at flashing the BIOS though

**Comment by Don-Keballs** [comment]
What are you using to flash the BIOS?

**Reply by SternXD** [reply]
CSME System Tools v11 r46

The part after you are supposed to patch it and get the rom.patched file is where I am confused. Because I don't think my board needs DSDT patching since it's not an ASRock board.

**Reply by Don-Keballs** [reply]
AFAIK You have to hardware flash these DELL BIOS with an external programmer like a CH341A or similar. Don't quote me though, I'm no expert. 

"External EEPROM flasher required. Dell BIOS flasher would not allow the flashing of modified BIOS."
https://github.com/xCuri0/ReBarUEFI/issues/11

**Reply by SternXD** [reply]
> AFAIK You have to hardware flash these DELL BIOS with an external programmer like a CH341A or similar. Don't quote me though, I'm no expert.
> 
> "External EEPROM flasher required. Dell BIOS flasher would not allow the flashing of modified BIOS." #11

That's for a different computer and in that same list there's the T3610 which didn't mention that at all. I can flash using CSME I just need to know what to do after the patching of the .rom file.

**Reply by SternXD** [reply]
So I have explored further and apparently there's this thing called boot guard which will not let my system POST if a modified BIOS is flashed.

**Comment by emailkanji** [comment]
Heyo. I did the "above 4g decoding" modification with probably success; more on that later.

I started by following the hidden 4g decoding guide on this github. It was a great resource for finding the hidden option. It unfortunately doesn't include the complexity of dell bios. I had to use GUID address as default used by set var tools isn't correct. Here is the info I used; found in bios 2.33.0.exe.

GUID: ACA9F304-21E2-4852-9875-7FF4881D67A5
VarStore Name: PCI_COMMON
VarOffset: 0x3
Total Size: 0x6 (6 bytes)

A read of an unedited 6 bytes from my device is as follows. I was confused at first because I thought offset 3 was the first byte but byte one is labelled 0.

0x0 00
0x1 00
0x2 01
0x3 00
0x4 00
0x5 00

Edited varoffset 0x3 to 01; entire 6 bytes were submitted because I didn't know how to use the tool well.

First boot appeared dead. Was warned this may occur. Left for 5 minutes or so in case it was doing something. Manually rebooted at least a couple of times. Panicked. And then unplugged computer and pulled cmos battery. Booted fine after that. Verified edit after cmos erase. No issue. But scary because I thought that would clear my setting in case it was the wrong setting.

Now for the probably part. Works fine with GPU removed. Everything I do with GPU installed goes to bios. Selecting boot menu or doing nothing results in bios. BIOS reports nvidia video card installed.

Will try larger PSU soonish; not optimistic as total system power never breaks 100 watts.

**Reply by SternXD** [reply]
I know when I tried it I had an RX 6650 XT installed I still have the PC but have since upgraded but that's awesome!
